### PR TITLE
[Dialogs] Update buttons touch area to be 48x48 minimum

### LINKED ISSUE #3664: [Dialogs] Alert button hit areas need to be at least 48 pt tall
They are currently 33 points tall.

### PR DESCRIPTION
**Pull request by codeman7**
Update the buttons within dialogs to respect 48x48 minimum touch area. 

Closes #3664

### CODE REVIEW

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m [hunk 1]
```diff
@@ -129,6 +129,12 @@ - (void)addActionButtonTitle:(NSString *)actionTitle target:(id)target selector:
   buttonRect.size.height = MAX(buttonRect.size.height, MDCDialogActionButtonHeight);
   buttonRect.size.width = MAX(buttonRect.size.width, MDCDialogActionButtonMinimumWidth);
   actionButton.frame = buttonRect;
+  CGFloat verticalInset = MIN(0, -(48 - CGRectGetHeight(actionButton.bounds)) / 2);
```
Comment (romoore) [inline components/Dialogs/src/private/MDCAlertControllerView+Private.m]: Are these automatically updated for dynamic type?

Comment (codeman7) [reply components/Dialogs/src/private/MDCAlertControllerView+Private.m]: The edge insets or the bounds of the buttons?

Comment (romoore) [reply components/Dialogs/src/private/MDCAlertControllerView+Private.m]: the edge insets. They should become much smaller if the button gets bigger due to dynamic type

Comment (codeman7) [reply components/Dialogs/src/private/MDCAlertControllerView+Private.m]: As far as I can see Dialogs don't support dynamic type

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m [hunk 2]
```diff
@@ -230,6 +230,17 @@ - (void)updateButtonFont {
   }
   for (MDCFlatButton *button in self.actionButtons) {
     [button setTitleFont:finalButtonFont forState:UIControlStateNormal];
+    [button sizeToFit];
+    CGRect buttonRect = button.frame;
+    buttonRect.size.height = MAX(CGRectGetHeight(buttonRect), MDCDialogActionButtonHeight);
+    buttonRect.size.width = MAX(CGRectGetWidth(buttonRect), MDCDialogActionButtonMinimumWidth);
+    button.frame = buttonRect;
+    CGFloat verticalInset = MIN(0, -(48 - CGRectGetHeight(button.bounds)) / 2);
+    CGFloat horizontalInset = MIN(0, -(48 - CGRectGetWidth(button.bounds)) / 2);
```
Comment (ianegordon) [inline components/Dialogs/src/private/MDCAlertControllerView+Private.m]: We can open an issue to deal with later, but 48 should likely be declared somewhere as something like MINIMUM_TOUCH_TARGET so 48 won't be magically scattered around our code.

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m [hunk 3]
```diff
@@ -230,6 +230,17 @@ - (void)updateButtonFont {
...
+    button.hitAreaInsets = UIEdgeInsetsMake(verticalInset,
+                                                  horizontalInset,
+                                                  verticalInset,
+                                                  horizontalInset);
```
Comment (ianegordon) [inline components/Dialogs/src/private/MDCAlertControllerView+Private.m]: If we are resizing view, we likely want to be dealing that code in layoutSubviews.  Generally, not just in this specific case.

At the top of layoutSubviews, where we are calling [button sizeToFit] on all the actions, I would add this code there.

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m [hunk 4]
```diff
@@ -230,6 +230,17 @@ - (void)updateButtonFont {
   }
   for (MDCFlatButton *button in self.actionButtons) {
     [button setTitleFont:finalButtonFont forState:UIControlStateNormal];
+    [button sizeToFit];
+    CGRect buttonRect = button.frame;
+    buttonRect.size.height = MAX(CGRectGetHeight(buttonRect), MDCDialogActionButtonHeight);
+    buttonRect.size.width = MAX(CGRectGetWidth(buttonRect), MDCDialogActionButtonMinimumWidth);
+    button.frame = buttonRect;
+    CGFloat verticalInset = MIN(0, -(48 - CGRectGetHeight(button.bounds)) / 2);
```
Comment (ianegordon) [inline components/Dialogs/src/private/MDCAlertControllerView+Private.m]: Can the second clause be simplified?
(CGRectGetHeight(button.bounds) - 48) / 2

Here an below.

**Review by ianegordon — DISMISSED** [review]
If you haven't already, could you verify the hit test for buttons works as expected in the Dynamic Type example for XXS as well as XXXL in horizontal and, if possible, vertical button layouts?

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m (hunk 1 above)
Comment (ianegordon) [reply components/Dialogs/src/private/MDCAlertControllerView+Private.m]: The view controller description is not great, but it was under Dialogs -> Alerts -> Alert (Dynamic).
I'll update the description so this doesn't happen again.

**Comment by jverkoey** [comment]
@ianegordon Do you have follow up feedback for this PR?

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m [hunk 5]
```diff
@@ -35,6 +35,7 @@
 static const CGFloat MDCDialogActionsVerticalPadding = 8.0;
 static const CGFloat MDCDialogActionButtonHeight = 36.0;
 static const CGFloat MDCDialogActionButtonMinimumWidth = 48.0;
+static const CGFloat MDCDialogActionButtonTouchTarget = 48.f;
```
Comment (ianegordon) [inline components/Dialogs/src/private/MDCAlertControllerView+Private.m]: super-nit:
Naming is hard.
Consider MDCDialogActionMinTouchTarget

File: components/Dialogs/src/private/MDCAlertControllerView+Private.m [hunk 6]
```diff
@@ -353,8 +355,24 @@ - (CGSize)calculateActionsSizeThatFitsWidth:(CGFloat)boundingWidth {
...
+  for (MDCFlatButton *button in self.actionButtons) {
     [button sizeToFit];
+    CGRect buttonFrame = button.frame;
+    buttonFrame.size.width =
+        MAX(CGRectGetWidth(buttonFrame), MDCDialogActionButtonMinimumWidth);
+    buttonFrame.size.height =
+        MAX(CGRectGetHeight(buttonFrame), MDCDialogActionButtonHeight);
+    button.frame = buttonFrame;
+    CGFloat verticalInsets =
+        -(MDCDialogActionButtonTouchTarget - CGRectGetHeight(button.frame)) / 2;
```
Comment (ianegordon) [inline components/Dialogs/src/private/MDCAlertControllerView+Private.m]: super-nit: Can this be simplified to here and below?
(CGRectGetHeight(button.frame) - MDCTouchTarget) / 2

**Review by ianegordon — APPROVED** [review]
Please consider comments before merge.

**Comment by romoore** [comment]
Looks like this introduced overlapping button touch targets in the "Overpopulated" Alert example.  Here's a print-out of the 6 vertically-stacked buttons. For each, you can see that the vertical touch targets overlap by 4 points.

Button 1: 2..50
Button 2: 46..94
Button 3: 90..138
Button 4: 134..182
Button 5: 178..226
Button 6: 222..270

```
2018-07-30 10:39:19.831 MDCCatalog[19231:215705] Button (0x79d81ee0) touch area: {{102.5, 2}, {75, 48}}
2018-07-30 10:39:19.831 MDCCatalog[19231:215705] Button (0x79f79720) touch area: {{102.5, 46}, {75, 48}}
2018-07-30 10:39:19.832 MDCCatalog[19231:215705] Button (0x79f641d0) touch area: {{102.5, 90}, {75, 48}}
2018-07-30 10:39:19.832 MDCCatalog[19231:215705] Button (0x79f701b0) touch area: {{102.5, 134}, {75, 48}}
2018-07-30 10:39:19.832 MDCCatalog[19231:215705] Button (0x79fb0340) touch area: {{102.5, 178}, {75, 48}}
2018-07-30 10:39:19.832 MDCCatalog[19231:215705] Button (0x79faa5f0) touch area: {{113.5, 222}, {53, 48}}
```

**Pull request merged**